sdk/core/amqp-common/src/ConnectionContextBase.ts

```diff
@@ -194,7 +198,17 @@ export module ConnectionContextBase {
           parameters.config.sharedAccessKeyName,
           parameters.config.sharedAccessKey
         ),
-      dataTransformer: parameters.dataTransformer || new DefaultDataTransformer()
+      dataTransformer: parameters.dataTransformer || new DefaultDataTransformer(),
+      refreshConnection() {
+        const connection = new Connection(connectionOptions);
```

Comment:
Just curious - why aren't you just directly assigning to this.connection and this.connectionLock?

It looks like you're avoiding something, just wondering what.

Reply:
my guess:
- coz he needs to use `connection.id` a few lines later, so he needs the connection to be a const/var
- I dont have an answer for the lock

Reply:
The lock was an oversight...it didn’t need its own variable.